### PR TITLE
 Create shortcuts

### PR DESCRIPTION
**Pull request by FabianDevel**
Depends on #1129

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt [hunk 1]
```diff
@@ -147,4 +148,12 @@ class LaunchActivity : AppCompatActivity() {
...
         }
     }
+
+    private fun handleShortcuts() {
+        intent.extras?.getString(SHORTCUTS_TAG)?.let { extrasMainActivity = MainActivityArgs(shortcutId = it).toBundle() }
+    }
+
+    private companion object {
+        const val SHORTCUTS_TAG = "shortcuts_tag"
+    }
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt:158]:
```suggestion
   companion object {
        private const val SHORTCUTS_TAG = "shortcuts_tag"
    }
```

Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt:158]:
```suggestion
    companion object {
        private const val SHORTCUTS_TAG = "shortcuts_tag"
    }
```

File: app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt [hunk 2]
```diff
@@ -30,13 +30,13 @@ interface IGeniusScanUtils {
...
-    fun Context.startScanFlow(resultLauncher: ActivityResultLauncher<Intent>) {
+    fun Activity.startScanFlow(resultLauncher: ActivityResultLauncher<Intent>) {
         MaterialAlertDialogBuilder(this, R.style.DialogStyle)
             .setTitle(R.string.allErrorFeatureNotAvailableInFdroid)
             .setPositiveButton("Ok") { _: DialogInterface?, _: Int -> }
             .show()
     }
 
-    fun Fragment.scanResultProcessing(intent: Intent, folder: File?) = Unit
+    fun Activity.scanResultProcessing(intent: Intent, folder: File?) = Unit
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt:40]: When do we put extensions in `Extensions.kt` file  and when we do not ?

**Review by tevincent — CHANGES_REQUESTED** [review]
Question

File: app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt (hunk 2 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt:40]: We put it in `Extension.kt` when the function is used in more than one file, but it hasn't its own Utils 
Creating a dedicated Utils for a functionality takes over all other files, so we regroup all functions linked to this functionality in the Utils, even if it's an extension

Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt:40]: This allows to keep related code in the same file, and also to reduce the `Extensions.kt size`, as it starts to get really big

**Review by tevincent — APPROVED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt [hunk 3]
```diff
@@ -43,7 +43,7 @@ import kotlinx.coroutines.withContext
 class LaunchActivity : AppCompatActivity() {
 
     private val navigationArgs: LaunchActivityArgs? by lazy { intent?.extras?.let { LaunchActivityArgs.fromBundle(it) } }
-    private var extrasOpenSpecificFile: Bundle? = null
+    private var extrasMainActivity: Bundle? = null
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt:46]: Test

**Review by tevincent — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt [hunk 4]
```diff
@@ -30,13 +30,13 @@ interface IGeniusScanUtils {
 
     fun Context.initGeniusScanSdk() = true
 
-    fun Context.startScanFlow(resultLauncher: ActivityResultLauncher<Intent>) {
+    fun Activity.startScanFlow(resultLauncher: ActivityResultLauncher<Intent>) {
         MaterialAlertDialogBuilder(this, R.style.DialogStyle)
             .setTitle(R.string.allErrorFeatureNotAvailableInFdroid)
             .setPositiveButton("Ok") { _: DialogInterface?, _: Int -> }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/utils/IGeniusScanUtils.kt:36]: Hardcoded string ? :(

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**